iXBRLViewerPlugin/viewer/src/js/util.js

```diff
@@ -73,10 +86,26 @@ export function wrapLabel(str, maxwidth){
         }
 
     });
-
     return sections;
 }
 
+/* The JSON format supports datetimes being abbreviated to just xsd:dates.
+ * moment.js doesn't support timezoned dates, so fix them to midnight before
+ * passing to moment */
+export function xbrlDateToMoment(dateString) {
+    /* If the string has something after the date part other than a time part,
+     * insert a time part of 'T00:00:00'
+     *
+     * i.e. 2010-01-01Z => 2010-01-01T00:00:00Z
+     *
+     */
+    dateString = dateString.replace(
+        /^(\d{4,}-\d{2}-\d{2})(?!T|$)/, 
+        function(match, $1) { return $1 + 'T00:00:00' }
```

Comment:
I don't understand how this works for instants or duration end dates.  If the value is `2019-03-31` in iXBRL, this will become `2019-03-31T00:00:00`, and then `momentToHuman` will render it as `2019-03-30`.  I believe this function similarly needs `adjust`, and then if there is no "T" (so a plain date) and `adjust` (so end date or instant), then it should `.add(1, 'day')` per
> A date, with no time part, in the endDate or instant element is defined to be equivalent to specifying a dateTime of the same date plus P1D and with a time part of T00:00:00.

Reply:
The trick is that we're not working with raw XBRL 2.1 dates.  We're working with dates from the JSON block which are truncated consistently to midnight at the start of the day.  This is consistent with the approach taken in xBRL-JSON: code and syntax should work consistently with datetimes, and the convention of adjusting instants and end dates should be applied purely with in the rendering layer (xBRL-JSON doesn't technically allow the removal of the time part at all).

Reply:
I see, thanks for clarifying.  I did not appreciate the difference in processing and likely would have been stuck for quite some time, so thanks for fixing this issue :-).